.github/scripts/compile.sh

```diff
@@ -106,32 +106,49 @@ do_build () {
   local suffix=${1}
   local runtime_check=${2}
 
+  if [ "$runtime_check" == "OFF" ]; then
+     # Turn on PGO only when we are not doing asan or tsan
+     BUILD_OPTS="-DENABLE_PROFILE_GUIDED_OPTIMIZATION=ON"
+  else
+     BUILD_OPTS="-DBUILD_TESTING=OFF"
+  fi
+
   cmake -S "${PROTON_DIR}" -B "${PROTON_BUILD_DIR}${suffix}" \
     -DCMAKE_BUILD_TYPE=RelWithDebInfo \
     -DRUNTIME_CHECK="${runtime_check}" \
     -DENABLE_LINKTIME_OPTIMIZATION=ON \
-    -DCMAKE_POLICY_DEFAULT_CMP0069=NEW -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON \
+    -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON \
     -DBUILD_TLS=ON -DSSL_IMPL=openssl -DBUILD_STATIC_LIBS=ON -DBUILD_BINDINGS=python \
     -DBUILD_EXAMPLES=OFF -DBUILD_TESTING=OFF \
     -DCMAKE_INSTALL_PREFIX=/usr
   cmake --build "${PROTON_BUILD_DIR}${suffix}" --verbose
 
   DESTDIR="$PROTON_INSTALL_DIR${suffix}" cmake --install "$PROTON_BUILD_DIR${suffix}"
 
+   if [ "$runtime_check" == "OFF" ]; then
+     # This will install the proton python libraries in sys.path so the tests using
+     # proton can be run successfully.
+     python3 -m pip install "$(find "$PROTON_BUILD_DIR/python/" -name 'python-qpid-proton-*.tar.gz')"
+   fi
+
   cmake -S "${SKUPPER_DIR}" -B "${SKUPPER_BUILD_DIR}${suffix}" \
     -DCMAKE_BUILD_TYPE=RelWithDebInfo \
     -DRUNTIME_CHECK="${runtime_check}" \
     -DProton_USE_STATIC_LIBS=ON \
     -DProton_DIR="${PROTON_INSTALL_DIR}${suffix}/usr/lib64/cmake/Proton" \
-    -DBUILD_TESTING=OFF \
+    ${BUILD_OPTS} \
     -DVERSION="${VERSION}" \
     -DCMAKE_INSTALL_PREFIX=/usr
   cmake --build "${SKUPPER_BUILD_DIR}${suffix}" --verbose
 }
 
+# This is required to install the python packages that the system tests use.
+python3 -m pip install -r "${SKUPPER_DIR}"/requirements-dev.txt
+
 # Do a regular build without asan or tsan.
 do_build "" OFF
-# and install Proton Python
+
+# Install Proton Python
 python3 -m pip install --prefix="$PROTON_INSTALL_DIR/usr" "$(find "$PROTON_BUILD_DIR/python/" -name 'python-qpid-proton-*.tar.gz')"
```

Comment:
```suggestion
python3 -m pip install --ignore-installed --prefix="$PROTON_INSTALL_DIR/usr" "$(find "$PROTON_BUILD_DIR/python/" -name 'python-qpid-proton-*.tar.gz')"
```

@ganeshmurthy this needs that `--ignore-installed` flag, as I had in https://github.com/jiridanek/skupper-router/actions/runs/5072575450/jobs/9110449383

Reply:
Forgot about that, adding it now.

Reply:
After I rebased with the podman changes, the ipv6 related errors have disappeared but I see this test failure - https://github.com/skupperproject/skupper-router/actions/runs/5084198540/jobs/9136182240?pr=1044#step:3:6634

I wonder if the CI is just slow and the 60 second timeout was hit but the same test did pass in other cases.

Reply:
> After I rebased with the podman changes, the ipv6 related errors have disappeared but I see this test failure - https://github.com/skupperproject/skupper-router/actions/runs/5084198540/jobs/9136182240?pr=1044#step:3:6634
> 
> I wonder if the CI is just slow and the 60 second timeout was hit but the same test did pass in other cases.

Regular CI jobs have 300 second timeout or so, set by environment variable. The tests running in Containerfile build have the default 60 second limit because there the variable is not set.

I tried to rettriger the CI job to see what happens. Defining the variable also for Containerfile might make sense.

Reply:
https://github.com/skupperproject/skupper-router/blob/4201a1ac7df591993b0082d26fa6e801daab1634/.github/workflows/build.yaml#LL459C21-L459C21

Reply:
Are you suggesting declaring the QPID_SYSTEM_TEST_TIMEOUT in the Containerfile like this - 

 ```diff
 ENV LIBUNWIND_VERSION=v1.6.2
 ENV LWS_SOURCE_URL=${LWS_SOURCE_URL:-https://github.com/warmcat/libwebsockets/archive/refs/tags/${LWS_VERSION}.tar.gz}
 ENV LIBUNWIND_SOURCE_URL=${LIBUNWIND_SOURCE_URL:-https://github.com/libunwind/libunwind/archive/refs/tags/${LIBUNWIND_VERSION}.tar.gz}
+ENV QPID_SYSTEM_TEST_TIMEOUT=300
 
 ARG VERSION=UNKNOWN
 ENV VERSION=$VERSION
``` 

Instead of doing in the Containerfile, we could also do this - 

```diff
-            COMMAND ${CMAKE_CTEST_COMMAND} -j4 -V -R tcp || (exit 0)
-            COMMAND ${CMAKE_CTEST_COMMAND} -j4 -V -R http || (exit 0)
+            COMMAND QPID_SYSTEM_TEST_TIMEOUT=300 ${CMAKE_CTEST_COMMAND} -j4 -V -R tcp || (exit 0)
+            COMMAND QPID_SYSTEM_TEST_TIMEOUT=300 ${CMAKE_CTEST_COMMAND} -j4 -V -R http || (exit 0)
```

Reply:
It seems to me that the extended timeout is necessary, otherwise the tests don't finish on CI, at least that's what the situation was when i last looked.

Whether to add it as Containerfile ENV or just for the command in CMakeLists.txt, I don't have a strong preference. I'd do the Containerfile ENV, but I see why the second option may look good too.

Reply:
> It seems to me that the extended timeout is necessary, otherwise the tests don't finish on CI, at least that's what the situation was when i last looked.
> 
> Whether to add it as Containerfile ENV or just for the command in CMakeLists.txt, I don't have a strong preference. I'd do the Containerfile ENV, but I see why the second option may look good too.

I will add the QPID_SYSTEM_TEST_TIMEOUT before the call to ctest. If there are too many tests that need using the QPID_SYSTEM_TEST_TIMEOUT, we will move it to the Containerfile